metrics/ci_integrations/test/source/buildkite/adapter/buildkite_source_client_adpter_test.dart

```diff
@@ -0,0 +1,1138 @@
+
+        whenFetchBuildkiteBuilds(
+          withArtifactsPage: defaultArtifactsPage,
+        ).thenSuccessWith(BuildkiteBuildsPage(values: [build]));
+
+        final firstBuild = testData.generateBuildData(buildNumber: 1);
+        final result = await adapter.fetchBuildsAfter(
+          testData.pipelineSlug,
+          firstBuild,
+        );
+        final duration = result.first.duration;
+
+        expect(duration, equals(expectedDuration));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() maps fetched builds' difference between the startedAt and finishedAt dates to the Duration.zero if the startedAt date is null",
+          () async {
+        final build = BuildkiteBuild(
+          number: 2,
+          blocked: false,
+          startedAt: null,
+          finishedAt: DateTime.now(),
+        );
+
+        whenFetchBuildkiteBuilds(
+          withArtifactsPage: defaultArtifactsPage,
+        ).thenSuccessWith(BuildkiteBuildsPage(values: [build]));
+
+        final firstBuild = testData.generateBuildData(buildNumber: 1);
+        final result = await adapter.fetchBuildsAfter(
+          testData.pipelineSlug,
+          firstBuild,
+        );
+        final duration = result.first.duration;
+
+        expect(duration, equals(Duration.zero));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() maps fetched builds' difference between the startedAt and finishedAt dates to the Duration.zero if the finishedAt date is null",
+          () async {
+        final build = BuildkiteBuild(
+          number: 2,
+          blocked: false,
+          startedAt: DateTime.now(),
+          finishedAt: null,
+        );
+
+        whenFetchBuildkiteBuilds(
+          withArtifactsPage: defaultArtifactsPage,
+        ).thenSuccessWith(BuildkiteBuildsPage(values: [build]));
+
+        final firstBuild = testData.generateBuildData(buildNumber: 1);
+        final result = await adapter.fetchBuildsAfter(
+          testData.pipelineSlug,
+          firstBuild,
+        );
+        final duration = result.first.duration;
+
+        expect(duration, equals(Duration.zero));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() maps fetched builds' url to the empty string if the url is null",
+          () async {
+        const build = BuildkiteBuild(
+          number: 2,
+          blocked: false,
+          webUrl: null,
+        );
+
+        whenFetchBuildkiteBuilds(
+          withArtifactsPage: defaultArtifactsPage,
+        ).thenSuccessWith(const BuildkiteBuildsPage(values: [build]));
+
+        final firstBuild = testData.generateBuildData(buildNumber: 1);
+        final result = await adapter.fetchBuildsAfter(
+          testData.pipelineSlug,
+          firstBuild,
+        );
+        final url = result.first.url;
+
+        expect(url, equals(''));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() maps fetched builds' startedAt date to the finishedAt date if the startedAt date is null",
```

Comment:
Should we add the test case `.fetchBuildsAfter() maps fetched builds' startedAt date to current date if the startedAt and finishedAt dates are null`? And the same for the `.fetchBuilds()` method?